app/atomic_svc.py

```diff
@@ -0,0 +1,240 @@
+        '''
+        Populate internal dictionary used to match techniques to corresponding tactics.
+        By default, use the file 'enterprise-attack.json' located in the Atomic Red Team repository.
+        You can use a specific version of this file by passing its path with the `entreprise_attack_path`
+        parameter.
+        '''
+        self.technique_to_tactics = defaultdict(list)
+        if not entreprise_attack_path:
+            entreprise_attack_path = os.path.join(self.repo_dir, 'atomic_red_team', 'enterprise-attack.json')
+
+        with open(entreprise_attack_path, 'r') as f:
+            mitre_json = json.load(f)
+
+        for phase_name, external_id in self._gen_single_match_tactic_technique(mitre_json):
+            self.technique_to_tactics[external_id].append(phase_name)
+
+    def _handle_attachment(self, attachment_path):
+        # attachment_path must be a POSIX path
+        payload_name = os.path.basename(attachment_path)
+        # to avoid collisions between payloads with the same name
+        with open(attachment_path, 'rb') as f:
+            h = hashlib.md5(f.read()).hexdigest()
+        payload_name = h[:6] + '_' + payload_name
+        shutil.copyfile(attachment_path, os.path.join(self.payloads_dir, payload_name), follow_symlinks=False)
+        return payload_name
+
+    @staticmethod
+    def _normalize_path(path, platform):
+        if platform == PLATFORMS['windows']:
+            return path.replace('\\', '/')
+        else:
+            return path
+
+    def _catch_path_to_atomics_folder(self, string, platform):
+        '''
+        Catch a path to the the atomics/ folder, and handle it in the best way
+        possible. If needed, will import a payload.
+        '''
+        regex = re.compile(r'\$?PathToAtomicsFolder((?:/[^/ \n]+)+|(?:\\[^\\ \n]+)+)')
+        payloads = []
+        if regex.search(string):
+            fullpath, path = regex.search(string).group(0, 1)
+            path = self._normalize_path(path, platform)
+
+            # take path from index 1, as it starts with /
+            path = os.path.join(self.repo_dir, 'atomics', path[1:])
+
+            # TODO handle folders
+            if os.path.isfile(path):
+                payload_name = self._handle_attachment(path)
+                payloads.append(payload_name)
+                string = string.replace(fullpath, payload_name)
+
+        return string, payloads
+
+    def _use_default_inputs(self, entries, test, platform, string):
+        payloads = []  # payloads induced by a variable
+        defaults = dict((key, val) for key, val in test.get('input_arguments', dict()).items())
+        while RE_VARIABLE.search(string):
+            full_var_string, varname = RE_VARIABLE.search(string).groups()
+            default_var = str(defaults.get(varname, dict()).get('default'))
+
+            if default_var:  # we found the default value of a variable
```

Comment:
comment on this line is redundant to the variable name (i.e., you named the variable good enough to not need a comment). remove comment.

Reply:
removed in 5b66cb5